packages/typescript-estree/src/create-program/useProvidedProgram.ts

```diff
@@ -0,0 +1,28 @@
+import debug from 'debug';
+import { Program } from 'typescript';
+import { Extra } from '../parser-options';
+import { ASTAndProgram, getAstFromProgram } from './shared';
+
+const log = debug('typescript-eslint:typescript-estree:useProvidedProgram');
+
+function useProvidedProgram(
+  programInstance: Program,
+  extra: Extra,
+): ASTAndProgram | undefined {
+  log('Retrieving ast for %s from provided program instance', extra.filePath);
+
+  const astAndProgram = getAstFromProgram(programInstance, extra);
+
+  if (!astAndProgram) {
+    const errorLines = [
+      '"parserOptions.program" has been provided for @typescript-eslint/parser.',
+      `The file was not found in the provided program instance: ${extra.filePath}`,
```

Comment:
The filepath should be made relative.
This ensures that error messages are kept is stable and portable.

https://github.com/typescript-eslint/typescript-eslint/blob/fc5f171b1ade2b45a1a4268b6d22926d420282a6/packages/typescript-estree/src/create-program/createProjectProgram.ts#L52-L55